src/passes/Directize.cpp

```diff
@@ -92,23 +104,29 @@ struct FunctionDirectizer : public WalkerPass<PostWalker<FunctionDirectizer>> {
 
 struct Directize : public Pass {
   void run(PassRunner* runner, Module* module) override {
+    TableUtils::FlatTable flatTable(module->table);
+    TableUtils::FlatTable* usedFlatTable = &flatTable;
```

Comment:
I think it would be clearer to make this a separate `bool` variable like `canOptimizeCallIndirects`.

Reply:
I found this confusing because it is largely redundant with the `flatTable` variable. It took me a bit to figure out that the only extra information `usedFlatTable` was encoding was whether the `flatTable` should be used. I believe that intent could be more simply (and therefore more clearly) represented as a bool. This seems different from other situations in which we use pointers to indicate optionality because in most of those situations, the optional pointee is either available or not present at all.

Reply:
Oh, I think we were talking about different things. Do you mean in this function itself, `run`? I agree it would make sense to add a bool there, I'll do that.